mlops/bicep/modules/azureml/attach_aks_training_to_azureml.bicep

```diff
@@ -0,0 +1,61 @@
+// This BICEP script will attach an AKS cluster
+// to a given AzureML workspace for training (NOT inferencing).
+
+// resource group must be specified as scope in az cli or module call
+targetScope = 'resourceGroup'
+
+// required parameters
+@description('Name of AzureML workspace to attach compute+storage to.')
+param machineLearningName string
+
+@description('The region of the machine learning workspace')
+param machineLearningRegion string = resourceGroup().location
+
+@description('Resource ID of the AKS cluster.')
+param aksResourceId string
+
+@description('Region of the AKS cluster.')
+param aksRegion string
+
+@description('How to name this compute in Azure ML')
+param amlComputeName string
+
+@description('Name of the existing UAI for the compute cluster.')
+param computeUaiName string
+
+// provision a user assigned identify for this silo
+resource uai 'Microsoft.ManagedIdentity/userAssignedIdentities@2022-01-31-preview' existing = {
+  name: computeUaiName
+  scope: resourceGroup()
+}
+
+var identityPrincipalId = uai.properties.principalId
+var userAssignedIdentities = {'/subscriptions/${subscription().subscriptionId}/resourceGroups/${resourceGroup().name}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${uai.name}': {}}
```

Comment:
just a nit, but I think we're doing this very same string interpolation at least 4 times, if I'm counting correctly. Do you know if there's a way we can introduce a custom function in bicep, that we could call in different bicep files? (I did a quick search and didn't find anything, but maybe you know a way already? If not, just drop it - this is neither critical nor expected to change)

Reply:
I think I might actually have found a way using resource() instead, which implicitely builds that whole path given a type and resource ID. I'll file an issue to implement in a next round.

Reply:
https://github.com/Azure-Samples/azure-ml-federated-learning/issues/143